client/patch-react.js

```diff
@@ -17,8 +17,23 @@ export default (handleError = () => {}) => {
   React.createElement = function (Component, ...rest) {
     if (typeof Component === 'function') {
       const { prototype } = Component
-      if (prototype && prototype.render) {
-        prototype.render = wrapRender(prototype.render)
+
+      // assumes it's a class component if render method exists.
+      const isClassComponent = Boolean(prototype && prototype.render) ||
+        // subclass of React.Component or PureComponent with no render method.
+        // There's no render method in prototype
+        // when it's created with class-properties.
+        prototype instanceof React.Component ||
+        prototype instanceof React.PureComponent
```

Comment:
Will this not make it work with class properties and alt-reacts like Preact? cc @developit